src/vdom/index.js

```diff
@@ -34,17 +34,10 @@ export function isNamedNode(node, nodeName) {
  * @returns {Object} props
  */
 export function getNodeProps(vnode) {
-	let props = extend({}, vnode.attributes);
-	props.children = vnode.children;
+	let props = vnode.children!==undefined ? { children: vnode.children } : {};
```

Comment:
instead of checking afterwards we can perform the check for the children before, and create the initial props either `{}` or `{children: vnode.children}`

The next line just takes previously created props as the first argument